src/libs/tools/include/errors/error.hpp

```diff
@@ -0,0 +1,70 @@
+#ifndef ELEKTRA_ERROR_HPP
+#define ELEKTRA_ERROR_HPP
+
+#include <vector>
+#include "errors/warning.hpp"
+
+namespace kdb
+{
+namespace tools
+{
+
+namespace errors
+{
+
+class Error : public BaseNotification
+{
+public:
+
+	/* inherit constructors */
+	using BaseNotification::BaseNotification;
+	virtual ~Error();
+
+	/**
+	 * @brief Add a warning to an error
+	 *
+	 * The warning is copied to make it independent from the source object. This way the same warning added to two different errors can be
+	 * changed independently.
+	 *
+	 * An Error can contain 0 to n warnings.
```

Comment:
A key can contain 0-n warnings. I would say that the error does.

Reply:
Here we adhered to the concept that was established by the Error-Keys and also the highlevel-API (`/src/libs/highlevel/elektra_error.c`).
Therefore we designed an Error-class that can hold multiple Warnings, so a Key of the C-API with 0 or 1 errors and 0 to n warnings can be directly converted to an object of type Error and the behaviour is consistent with the highlevel-API.

I extended the comment to explain the concept a bit in commit c2ec7ebbb6bbddfda4f82b9d9e44aee27d9971e2.